aten/src/ATen/ParallelCommon.cpp

```diff
@@ -47,10 +64,31 @@ std::string get_parallel_info() {
      << std::thread::hardware_concurrency() << std::endl;
 
   ss << "Environment variables:" << std::endl;
-  ss << "\tOMP_NUM_THREADS : " << get_env_var("OMP_NUM_THREADS") << std::endl;
-  ss << "\tMKL_NUM_THREADS : " << get_env_var("MKL_NUM_THREADS") << std::endl;
+  ss << "\tOMP_NUM_THREADS : "
+     << get_env_var("OMP_NUM_THREADS", "[not set]") << std::endl;
+  ss << "\tMKL_NUM_THREADS : "
+     << get_env_var("MKL_NUM_THREADS", "[not set]") << std::endl;
+
+  ss << "Parallel backend: ";
```

Comment:
Maybe instead of doing an if elif you could print all possibilities and then which one was chosen. We could have multiple switched on at once right?

Reply:
no, the ones below are mutually exclusive, unfortunately
I'll print all possibilities